docs/example/example/processes/template_py_process.py

```diff
@@ -0,0 +1,277 @@
+"""Describe what tasks this Process accomplishes."""
+# Order of imported packages should be:
+# future (Python 2)
+# Standard libraries
+from pathlib import Path
+from shutil import copy2
+
+# Third party libraries
+from plumbum import TEE
+
+# resolwe
+from resolwe.process import (
+    Process,
+    SchedulingClass,
+    Persistence,
+    Cmd,
+    StringField,
+    TextField,
+    BooleanField,
+    IntegerField,
+    FloatField,
+    DateField,
+    DateTimeField,
+    UrlField,
+    SecretField,
+    FileField,
+    FileHtmlField,
+    DirField,
+    JsonField,
+    ListField,
+    DataField,
+    GroupField,
+)
+
+# In-house libraries (i.e. resolwe_bio)
+# Local libraries
+
+
+class MergeReads(Process):
+    """Process description in one line ends with a period.
+
+    Continue with a more elaborate description of what the process
+    does, short layout of the algorithm and what output(s) to expect.
+
+    Note that this is not a working process, but rather just shows
+    bits and pieces that can be used in your real process.
+    """
+
+    slug = "merge-reads"
+    name = "Merged reads"  # human readable name of the process
+    process_type = "data:reads:merged:"
+    version = "1.0.0"
+    # See a list of existing categories in resolwe-bio documentation
+    # https://resolwe-bio.readthedocs.io/en/latest/catalog-category.html
+    category = "Read Processing"
+    # Dynamically generated name of the resulting datum object. Lookup
+    # happens from Input class fields.
+    data_name = '{{ string_field.file|default("?") }}'
+    scheduling_class = SchedulingClass.BATCH  # or .INTERACTIVE for short jobs
+    # or RAW or TMP, see documentation at
+    # https://resolwe.readthedocs.io/en/latest/proc.html#persistence
+    persistence = Persistence.CACHED
+    # Use below entity to make the created object also a sample.
+    # https://resolwe.readthedocs.io/en/latest/proc.html#entity
+    # descriptor_schema, input and always_create are optional.
+    entity = {
+        "type": "sample",  # name of your entity
+        # Use a descriptor schema slug if it differs from "type".
+        "descriptor_schema": "descriptor-schema-of-entity",
+        "input": "name of input that has the entity",
+        # If you want to create entity in all cases
+        "always_create": True,
+    }
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                # Refer to a valid Docker image. If using process
+                # within resolwe-bio, note that explicit version number
+                # as 'latest' is not permitted.
+                "image": "resolwebio/common:1.3.1"
+            }
+        },
+        "resources": {
+            "cores": 8,
+            "memory": 16384,
+            # Connects container to a network.
+            "network": True,
+        },
+    }
+
+    class Input:
+        """Input fields to process MergeData."""
+
+        string_field = StringField(
```

Comment:
There is a lot of other options, e.g. `allow_custom_choice`, `custom_choice_conditions`, `validate_regex`,...

Reply:
Listing all of them for every Field would become rather verbose, especially for fields that are not used often. This would perhaps be better suited to be listed in each Field's documentation page.

I have added a few new arguments, like `placeholder`, `range` and `validate_regex` to a few entries.

Reply:
Not listing all options for each field, but it would be nice to make one (or two) fields more verbose and list all available options.

Reply:
I believe `placeholder`, `range` and `validate_regex` do not work in Python processes yet so I've removed them. I have also made `string_field` more verbose as it includes all possible arguments.